### ISSUE TITLE
Bug: False positive `comma-dangle` when linting multiline operators

### ISSUE THREAD

**Issue by davidhcefx** [eslint/eslint#15885]
### Environment

Node version: v14.16.1
npm version: v8.3.0
Local ESLint version: v8.15.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19044

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    commonjs: true,
    es2021: true,
  },
  extends: [
    'airbnb-base',
  ],
  parserOptions: {
    ecmaVersion: 'latest',
  },
  rules: {
    'no-console': 'off',
  },
};
```
</details>

The line with the multiline string produced a `  7:25  error  Missing trailing comma  (comma-dangle)`. However, it's just a multiline string rather than multiline arguments.

```js
function greet(names) {
  const arr = [];

  names.forEach((name) => {
    arr.push(
      `Hello ${name}, `
      + 'how are you?'
    );
  });
  console.log(arr.join('\n'));
}

greet(['David', 'Emilia']);
```

[Airbnb style guide][] also stated that if a control statement gets too long, each condition should be on its own line, which in some degree supported the above style. In addition, changing it to a Boolean condition also produced the same linter error.

```js
arr.push(
  name === 'David'
  || name === 'Emilia'
);
```

[Airbnb style guide]: https://github.com/airbnb/javascript#:~:text=In%20case%20your%20control%20statement%20(if%2C%20while%20etc.)%20gets%20too%20long%20or%20exceeds%20the%20maximum%20line%20length%2C%20each%20(grouped)%20condition%20could%20be%20put%20into%20a%20new%20line.

### What did you expect to happen?

No `comma-dangle` error

### What actually happened?

`  7:25  error  Missing trailing comma  (comma-dangle)`

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I've also tried to replicate it on [ESLint demo][] but failed. I don't know why.

[ESLint demo]: https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gZ3JlZXQobmFtZXMpIHtcbiAgY29uc3QgYXJyID0gW107XG5cbiAgbmFtZXMuZm9yRWFjaCgobmFtZSkgPT4ge1xuICAgIGFyci5wdXNoKFxuICAgICAgYEhlbGxvICR7bmFtZX0sIGBcbiAgICAgICsgJ2hvdyBhcmUgeW91PydcbiAgICApO1xuICB9KTtcbiAgY29uc29sZS5sb2coYXJyLmpvaW4oJ1xcbicpKTtcbn1cblxuZ3JlZXQoWydEYXZpZCcsICdFbWlsaWEnXSk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImFjY2Vzc29yLXBhaXJzIjoyLCJhcnJheS1icmFja2V0LW5ld2xpbmUiOjIsImFycmF5LWJyYWNrZXQtc3BhY2luZyI6MiwiYXJyYXktY2FsbGJhY2stcmV0dXJuIjoyLCJhcnJvdy1ib2R5LXN0eWxlIjoyLCJhcnJvdy1wYXJlbnMiOjIsImFycm93LXNwYWNpbmciOjIsImJsb2NrLXNjb3BlZC12YXIiOjIsImJsb2NrLXNwYWNpbmciOjIsImJyYWNlLXN0eWxlIjoyLCJjYW1lbGNhc2UiOjIsImNhcGl0YWxpemVkLWNvbW1lbnRzIjoyLCJjbGFzcy1tZXRob2RzLXVzZS10aGlzIjoyLCJjb21tYS1zcGFjaW5nIjoyLCJjb21tYS1zdHlsZSI6MiwiY29tcGxleGl0eSI6MiwiY29tcHV0ZWQtcHJvcGVydHktc3BhY2luZyI6MiwiY29uc2lzdGVudC1yZXR1cm4iOjIsImNvbnNpc3RlbnQtdGhpcyI6MiwiY29uc3RydWN0b3Itc3VwZXIiOjIsImN1cmx5IjoyLCJkZWZhdWx0LWNhc2UiOjIsImRlZmF1bHQtY2FzZS1sYXN0IjoyLCJkZWZhdWx0LXBhcmFtLWxhc3QiOjIsImRvdC1sb2NhdGlvbiI6MiwiZG90LW5vdGF0aW9uIjoyLCJlb2wtbGFzdCI6MiwiZXFlcWVxIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJmdW5jLWNhbGwtc3BhY2luZyI6MiwiZnVuYy1uYW1lLW1hdGNoaW5nIjoyLCJmdW5jLW5hbWVzIjoyLCJmdW5jdGlvbi1jYWxsLWFyZ3VtZW50LW5ld2xpbmUiOjIsImdlbmVyYXRvci1zdGFyLXNwYWNpbmciOjIsImdldHRlci1yZXR1cm4iOjIsImdyb3VwZWQtYWNjZXNzb3ItcGFpcnMiOjIsImd1YXJkLWZvci1pbiI6MiwiaWQtZGVueWxpc3QiOjIsImlkLWxlbmd0aCI6MiwiaWQtbWF0Y2giOjIsImltcGxpY2l0LWFycm93LWxpbmVicmVhayI6MiwiaW5pdC1kZWNsYXJhdGlvbnMiOjIsImpzeC1xdW90ZXMiOjIsImtleS1zcGFjaW5nIjoyLCJrZXl3b3JkLXNwYWNpbmciOjIsImxpbmUtY29tbWVudC1wb3NpdGlvbiI6MiwibGluZWJyZWFrLXN0eWxlIjoyLCJsaW5lcy1hcm91bmQtY29tbWVudCI6MiwibGluZXMtYmV0d2Vlbi1jbGFzcy1tZW1iZXJzIjoyLCJtYXgtY2xhc3Nlcy1wZXItZmlsZSI6MiwibWF4LWRlcHRoIjoyLCJtYXgtbGVuIjoyLCJtYXgtbGluZXMiOjIsIm1heC1saW5lcy1wZXItZnVuY3Rpb24iOjIsIm1heC1uZXN0ZWQtY2FsbGJhY2tzIjoyLCJtYXgtcGFyYW1zIjoyLCJtYXgtc3RhdGVtZW50cyI6MiwibWF4LXN0YXRlbWVudHMtcGVyLWxpbmUiOjIsIm11bHRpbGluZS1jb21tZW50LXN0eWxlIjoyLCJtdWx0aWxpbmUtdGVybmFyeSI6MiwibmV3LWNhcCI6MiwibmV3LXBhcmVucyI6MiwibmV3bGluZS1wZXItY2hhaW5lZC1jYWxsIjoyLCJuby1hbGVydCI6Miwibm8tYXJyYXktY29uc3RydWN0b3IiOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWF3YWl0LWluLWxvb3AiOjIsIm5vLWJpdHdpc2UiOjIsIm5vLWNhbGxlciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbmZ1c2luZy1hcnJvdyI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1iaW5hcnktZXhwcmVzc2lvbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb25zdHJ1Y3Rvci1yZXR1cm4iOjIsIm5vLWNvbnRpbnVlIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZGl2LXJlZ2V4IjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1kdXBsaWNhdGUtaW1wb3J0cyI6Miwibm8tZWxzZS1yZXR1cm4iOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LWZ1bmN0aW9uIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1lcS1udWxsIjoyLCJuby1ldmFsIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dGVuZC1uYXRpdmUiOjIsIm5vLWV4dHJhLWJpbmQiOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtbGFiZWwiOjIsIm5vLWV4dHJhLXBhcmVucyI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZsb2F0aW5nLWRlY2ltYWwiOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBsaWNpdC1jb2VyY2lvbiI6Miwibm8taW1wbGljaXQtZ2xvYmFscyI6Miwibm8taW1wbGllZC1ldmFsIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbmxpbmUtY29tbWVudHMiOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWludmFsaWQtdGhpcyI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLWl0ZXJhdG9yIjoyLCJuby1sYWJlbC12YXIiOjIsIm5vLWxhYmVscyI6Miwibm8tbG9uZS1ibG9ja3MiOjIsIm5vLWxvbmVseS1pZiI6Miwibm8tbG9vcC1mdW5jIjoyLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6Miwibm8tbWFnaWMtbnVtYmVycyI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLW9wZXJhdG9ycyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1tdWx0aS1hc3NpZ24iOjIsIm5vLW11bHRpLXNwYWNlcyI6Miwibm8tbXVsdGktc3RyIjoyLCJuby1tdWx0aXBsZS1lbXB0eS1saW5lcyI6Miwibm8tbmVnYXRlZC1jb25kaXRpb24iOjIsIm5vLW5lc3RlZC10ZXJuYXJ5IjoyLCJuby1uZXciOjIsIm5vLW5ldy1mdW5jIjoyLCJuby1uZXctb2JqZWN0IjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1uZXctd3JhcHBlcnMiOjIsIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1vY3RhbC1lc2NhcGUiOjIsIm5vLXBhcmFtLXJlYXNzaWduIjoyLCJuby1wbHVzcGx1cyI6Miwibm8tcHJvbWlzZS1leGVjdXRvci1yZXR1cm4iOjIsIm5vLXByb3RvIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1yZXN0cmljdGVkLWV4cG9ydHMiOjIsIm5vLXJlc3RyaWN0ZWQtZ2xvYmFscyI6Miwibm8tcmVzdHJpY3RlZC1pbXBvcnRzIjoyLCJuby1yZXN0cmljdGVkLXByb3BlcnRpZXMiOjIsIm5vLXJlc3RyaWN0ZWQtc3ludGF4IjoyLCJuby1yZXR1cm4tYXNzaWduIjoyLCJuby1yZXR1cm4tYXdhaXQiOjIsIm5vLXNjcmlwdC11cmwiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZWxmLWNvbXBhcmUiOjIsIm5vLXNlcXVlbmNlcyI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93IjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGFicyI6Miwibm8tdGVtcGxhdGUtY3VybHktaW4tc3RyaW5nIjoyLCJuby10ZXJuYXJ5IjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdGhyb3ctbGl0ZXJhbCI6Miwibm8tdHJhaWxpbmctc3BhY2VzIjoyLCJuby11bmRlZiI6Miwibm8tdW5kZWYtaW5pdCI6Miwibm8tdW5kZWZpbmVkIjoyLCJuby11bmRlcnNjb3JlLWRhbmdsZSI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVubW9kaWZpZWQtbG9vcC1jb25kaXRpb24iOjIsIm5vLXVubmVlZGVkLXRlcm5hcnkiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnJlYWNoYWJsZS1sb29wIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOjIsIm5vLXVudXNlZC1leHByZXNzaW9ucyI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXByaXZhdGUtY2xhc3MtbWVtYmVycyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZS1iZWZvcmUtZGVmaW5lIjoyLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOjIsIm5vLXVzZWxlc3MtY2FsbCI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1jb21wdXRlZC1rZXkiOjIsIm5vLXVzZWxlc3MtY29uY2F0IjoyLCJuby11c2VsZXNzLWNvbnN0cnVjdG9yIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8tdXNlbGVzcy1yZW5hbWUiOjIsIm5vLXVzZWxlc3MtcmV0dXJuIjoyLCJuby12YXIiOjIsIm5vLXZvaWQiOjIsIm5vLXdhcm5pbmctY29tbWVudHMiOjIsIm5vLXdoaXRlc3BhY2UtYmVmb3JlLXByb3BlcnR5IjoyLCJuby13aXRoIjoyLCJub25ibG9jay1zdGF0ZW1lbnQtYm9keS1wb3NpdGlvbiI6Miwib2JqZWN0LWN1cmx5LW5ld2xpbmUiOjIsIm9iamVjdC1jdXJseS1zcGFjaW5nIjoyLCJvYmplY3QtcHJvcGVydHktbmV3bGluZSI6Miwib2JqZWN0LXNob3J0aGFuZCI6Miwib25lLXZhciI6Miwib25lLXZhci1kZWNsYXJhdGlvbi1wZXItbGluZSI6Miwib3BlcmF0b3ItYXNzaWdubWVudCI6MiwicGFkZGluZy1saW5lLWJldHdlZW4tc3RhdGVtZW50cyI6MiwicHJlZmVyLWFycm93LWNhbGxiYWNrIjoyLCJwcmVmZXItY29uc3QiOjIsInByZWZlci1kZXN0cnVjdHVyaW5nIjoyLCJwcmVmZXItZXhwb25lbnRpYXRpb24tb3BlcmF0b3IiOjIsInByZWZlci1uYW1lZC1jYXB0dXJlLWdyb3VwIjoyLCJwcmVmZXItbnVtZXJpYy1saXRlcmFscyI6MiwicHJlZmVyLW9iamVjdC1oYXMtb3duIjoyLCJwcmVmZXItb2JqZWN0LXNwcmVhZCI6MiwicHJlZmVyLXByb21pc2UtcmVqZWN0LWVycm9ycyI6MiwicHJlZmVyLXJlZ2V4LWxpdGVyYWxzIjoyLCJwcmVmZXItcmVzdC1wYXJhbXMiOjIsInByZWZlci1zcHJlYWQiOjIsInByZWZlci10ZW1wbGF0ZSI6MiwicXVvdGUtcHJvcHMiOjIsInJhZGl4IjoyLCJyZXF1aXJlLWF0b21pYy11cGRhdGVzIjoyLCJyZXF1aXJlLWF3YWl0IjoyLCJyZXF1aXJlLXVuaWNvZGUtcmVnZXhwIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJyZXN0LXNwcmVhZC1zcGFjaW5nIjoyLCJzZW1pIjoyLCJzZW1pLXNwYWNpbmciOjIsInNlbWktc3R5bGUiOjIsInNvcnQtaW1wb3J0cyI6Miwic29ydC1rZXlzIjoyLCJzb3J0LXZhcnMiOjIsInNwYWNlLWJlZm9yZS1ibG9ja3MiOjIsInNwYWNlLWluLXBhcmVucyI6Miwic3BhY2UtaW5maXgtb3BzIjoyLCJzcGFjZS11bmFyeS1vcHMiOjIsInNwYWNlZC1jb21tZW50IjoyLCJzd2l0Y2gtY29sb24tc3BhY2luZyI6Miwic3ltYm9sLWRlc2NyaXB0aW9uIjoyLCJ0ZW1wbGF0ZS1jdXJseS1zcGFjaW5nIjoyLCJ0ZW1wbGF0ZS10YWctc3BhY2luZyI6MiwidW5pY29kZS1ib20iOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyLCJ2YXJzLW9uLXRvcCI6Miwid3JhcC1paWZlIjoyLCJ3cmFwLXJlZ2V4IjoyLCJ5aWVsZC1zdGFyLXNwYWNpbmciOjIsInlvZGEiOjIsImNvbW1hLWRhbmdsZSI6Mn0sImVudiI6eyJjb21tb25qcyI6dHJ1ZSwiYnJvd3NlciI6dHJ1ZSwiZXMyMDIxIjp0cnVlfX19

**Comment by mdjermanovic** [comment]
Hi @davidhcefx, thanks for the issue!

This works as intended, as per the [documentation](https://eslint.org/docs/rules/comma-dangle) for the `comma-dangle` rule:

> * `"always-multiline"` requires trailing commas when the last element or property is in a *different* line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the *same* line as the closing `]` or `}`

In the provided example, the argument is in a different line than the closing parenthesis, and therefore the rule requires a trailing comma after the argument. The rationale is that it improves the clarity of diffs when additional arguments are added, as explained in the first section of the document and in [airbnb style guide - Additional trailing comma](https://github.com/airbnb/javascript#commas--dangling).
